### DISCUSSION TITLE
QField refus to restart after connecting GPS

### DISCUSSION THREAD

**Discussion by FloCuf** [opengisch/QField#4846] [Q&A]
This bug started after the update "3.1.0. Borneo"
I use a Leica FLX 100 antenna managed with Leica Zeno connect.
After I configurated the bluetooth connection in QField, I connected it and everthing is ok but if I close the app, it is impossible to reopen it. After 1 or 2 seconds, it shutdown without any message.

In waiting for this correction, is it possible to downgrade to 3.0 Amazonia? how to do it?

**Comment by nirvn** [comment]
@FloCuf , what phone model do you have?

**Reply by FloCuf** [reply]
Xiaomi 11 Lite 5G  model 2109119DG  on Android 13

**Reply by nirvn** [reply]
@FloCuf , could you install QField 3.1.2 via the APK in the release page ( https://github.com/opengisch/QField/releases/tag/v3.1.2 ) and report whether the crash is gone?

**Reply by nirvn** [reply]
@FloCuf , direct link to the APK in case you are wondering: https://github.com/opengisch/QField/releases/download/v3.1.2/qfield-v3.1.2-arm64-android.apk

Also, could you make sure that you answer ( yes ) when asked whether to share anonymous metrics? That'll help us capture details of your crash.

**Reply by FloCuf** [reply]
Ok, well instaled, I just did it and it crashed again. Well shared anonymous metrics.

**Reply by nirvn** [reply]
@FloCuf , the issue has been replicated locally, thanks, expect a fix soon.

**Reply by FloCuf** [reply]
Great! Thank you

**Reply by nirvn** [reply]
@FloCuf , can you confirm that this APK (https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4847-d42115f5a6830b8108747a8f0665571ea9a9c026-arm64-android.apk) fixes the crash for you?

**Reply by FloCuf** [reply]
It's fixed indeed. I keep this one until the next upload ?

**Reply by nirvn** [reply]
@FloCuf , you definitively can. Note that the app settings and project data between the normal QField releases you find on the play store and manual installs of a PR build are separate. This means project data will not be shared across the two instances. If you use QFieldCloud, it's not really a problem as you can just push things to the cloud then re-download using the next official QField version :)

Thanks for alerting us of this issue and confirming the code changed has fixed the regression.
